src/coefficients/MFEMConstantCoefficient.C

```diff
@@ -12,7 +13,8 @@ MFEMConstantCoefficient::validParams()
 
 MFEMConstantCoefficient::MFEMConstantCoefficient(const InputParameters & parameters)
   : MFEMCoefficient(parameters),
-    coefficient{std::make_shared<mfem::ConstantCoefficient>(getParam<double>("value"))}
+    coefficient{getMFEMProblem().getProblemData()._scalar_manager.make<mfem::ConstantCoefficient>(
+        getParam<double>("value"))}
```

Comment:
Would it be worth adding a `addScalarCoefficient<T>(args...)` method to `MFEMProblem` that wraps the `getProblemData()._scalar_manager.make<T>(args...)` call for readability?

I understand we're expecting to remove the `MFEMCoefficient` classes as objects that end users can create from the input stack directly, but I suspect developers may still want to create (tracked) coefficients in some kernels/BCs/Executioners, where such a method on `MFEMProblem` would help.

(Similarly for vector and matrix coefficients)

Reply:
I've created those convenience methods. They will be useful for the new boundary condition classes in the next PR.